### PR TITLE
Option for extraDtab in Namerd handleGetDelegate

### PR DESCRIPTION
**Pull request by Ashald**
Follow up for PR #529

### CODE REVIEW

File: namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala [hunk 1]
```diff
@@ -397,12 +397,13 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
...
   }
 
-  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: String): Future[Response] = {
+  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: Option[String]): Future[Response] = {
     getDtab(ns).flatMap {
       case Some(dtab) =>
         delegate(ns) match {
           case delegator: Delegator =>
-            DelegateApiHandler.getDelegateRsp((dtab.dtab ++ Dtab.read(extraDtab)).show, path.show, delegator)
+            val fullDtab = extraDtab match { case Some(str) => dtab.dtab ++ Dtab.read(str); case _ => dtab.dtab }
```
Comment (adleong) [inline namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala]: I'd break up this line:
``
val fullDtab = extraDtab match {
  case Some(str) => ...
  case None => ...
}
```

Comment (Ashald) [inline namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala]: Done. ;)

File: namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala [hunk 2]
```diff
@@ -397,12 +397,16 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
...
 
-  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: String): Future[Response] = {
+  private[this] def handleGetDelegate(ns: String, path: Path, extraDtab: Option[String]): Future[Response] = {
     getDtab(ns).flatMap {
       case Some(dtab) =>
         delegate(ns) match {
           case delegator: Delegator =>
-            DelegateApiHandler.getDelegateRsp((dtab.dtab ++ Dtab.read(extraDtab)).show, path.show, delegator)
+            val fullDtab = extraDtab match {
+              case Some(str) => dtab.dtab ++ Dtab.read(str);
```
Comment (adleong) [inline namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala]: style nit: omit semicolon

Comment (Ashald) [inline namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala]: Done

**Comment by adleong** [comment]
:star: Awesome, thanks!

**Pull request merged**